### PR TITLE
add nicmacaddresslist parameter to importUnmanagedInstance and importVM APIs

### PR DESCRIPTION
**Pull request by prashanthr2**
## Description
Adds support for specifying per-NIC MAC addresses when importing a VM into CloudStack via `importUnmanagedInstance` and `importVM` APIs.

## Problem
Both APIs always used the MAC address reported by the hypervisor. The only MAC control was the `forced` boolean — which either rejected a conflict or silently generated a new MAC. There was no way to supply a specific MAC at import time, which is needed for migration scenarios requiring MAC preservation.

## Solution
New MAP parameter `nicmacaddresslist` following the same pattern as `nicnetworklist` and `nicipaddresslist`:

\`\`\`
nicmacaddresslist[0].nic=nic1&nicmacaddresslist[0].mac=aa:bb:cc:dd:ee:ff
nicmacaddresslist[1].nic=nic2&nicmacaddresslist[1].mac=aa:bb:cc:dd:ee:02
\`\`\`

NICs not listed keep their hypervisor-reported MAC — fully backward compatible.

## Changes
- `ApiConstants`: add `NIC_MAC_ADDRESS_LIST = "nicmacaddresslist"`
- `VmDetailConstants`: add `NIC_MAC_ADDRESS = "macAddress"`
- `ImportUnmanagedInstanceCmd`: new `@Parameter nicmacaddresslist` with `getNicMacAddressList()` — `isValidMac` + `isUnicastMac` + `standardizeMacAddress` validation
- `UnmanagedVMsManagerImpl`:
  - `mergeNicMacAddresses()`: merges caller MACs into the `IpAddresses` map
  - `baseImportInstance()` + `importVm()`: call `mergeNicMacAddresses()`
  - `importNic()`: prefers caller-supplied MAC over hypervisor-reported MAC
  - `importNic()`: `private` → `protected` for testability

## Tests
- `ImportUnmanagedInstanceCmdTest` (new): 9 tests — MAC format, unicast, empty/null validation
- `UnmanagedVMsManagerImplTest`: 8 new tests — `mergeNicMacAddresses()` edge cases + `importNic()` MAC selection

### CODE REVIEW

File: api/src/main/java/org/apache/cloudstack/api/command/admin/vm/ImportUnmanagedInstanceCmd.java [hunk 1]
```diff
@@ -135,6 +135,14 @@ public class ImportUnmanagedInstanceCmd extends BaseAsyncCmd {
             description = "VM NIC to ip address mapping using keys NIC, ip4Address")
     private Map nicIpAddressList;
 
+    @Parameter(name = ApiConstants.NIC_MAC_ADDRESS_LIST,
+            type = CommandType.MAP,
+            description = "VM NIC to MAC address mapping using keys nic and macAddress. " +
+                    "Example: nicmacaddresslist[0].nic=nic1&nicmacaddresslist[0].macAddress=aa:bb:cc:dd:ee:ff. " +
+                    "Overrides the MAC address reported by the hypervisor for the given NIC.",
+            since = "4.21.0")
```
Comment (weizhouapache) [inline api/src/main/java/org/apache/cloudstack/api/command/admin/vm/ImportUnmanagedInstanceCmd.java:143]: 4.23.0 ?

**Comment by weizhouapache** [comment]
@blueorangutan package

**Comment by weizhouapache** [comment]
moving to 4.24.0 milestone
